### ISSUE TITLE
tween by speed, not duration

### ISSUE THREAD

**Issue by marcusx2** [playcanvas/playcanvas-tween#25]
It would be nice if we could pass a speed parameter for tweening instead of duration. It would be useful if you want to move from point x to point y at a constant speed regardless of the distance of said points. With only the duration, a movement tween can happen faster or slower depending on the distance between point x and point y.

**Comment by yaustar** [comment]
Going at a constant speed isn't a tween. Tweening in animation is a short for inbetweening and it's the process of generating images that go between keyframes.

Keyframes are on a timeline so having an option to tween by speed doesn't make sense for a tweening library.

**Comment by marcusx2** [comment]
Well I've seen plenty of tweening libraries with this option before, even if it's not technically tweening.

**Comment by yaustar** [comment]
Can you link to them please? It will help with working out what API people are used to if we do implement something like this.

**Comment by marcusx2** [comment]
Well if the user provides the speed you can get the time with `time = (distance / speed) * dt;` right? The distance variable will change based on what you are trying to tween. If it's a movement tween distance is really the distance between two vectors, but if you are trying to tween, idk, the color of something, distance will be something else. But that's the gist of it.

Hm let's see an example [http://dotween.demigiant.com/documentation.php](http://dotween.demigiant.com/documentation.php). Search for `Tweener-specific settings and options` and then look into the SetSpeedBased option.

**Comment by yaustar** [comment]
Yeah, this is the part where I feel like it doesn't belong in a tween library:

![image](https://user-images.githubusercontent.com/16639049/143250613-38078473-83ae-407d-b8d5-e54ab8ac4f09.png)

It basically becomes a standard lerp as it the easing option are basically ignored which is the main feature.

I can see a need for it all that said so consideration will be given to add this or consider a PR

**Comment by marcusx2** [comment]
>as it the easing option are basically ignored which is the main feature.

Only if the interpolator is linear.
